src/compas_occ/brep/brep.py

```diff
@@ -148,6 +148,11 @@ class BRep(Data):
 
     """
 
+    # this makes de-serialization backend-agnostic.
+    # The deserializing type is determined by plugin availability when de-serializing
+    # regardless of the context available when serializing.
+    __class__ = BrepPluggable
```

Comment:
not sure why this is necessary

Reply:
is this name also not a bit confusing?

would it not be better in general to give the plugin a specific name, such as `RhinoBrep` and `OCCBrep`? (which would allow `__class__ = Brep`)

i am aware of the fact that this is not consistently implemented everywhere, but perhaps it should be...

Reply:
This is done so that the `dtype` for a serialized Brep will be `compas.geometry/Brep` rather than `compas_occ.brep.BRep` so that, when de-serializing, whatever backend is currently available shall handle the de-serialization not relying on the backend that did the serialization.

```json
{
	"dtype": "compas.geometry/Brep",
	"value": {
		"faces": []
```

For the naming, yes I fully agree. The idea was to make the minimum amount of changes required to make this a backend. But for sure, I agree we should standardize.